include/glow/Optimizer/GraphOptimizer/FunctionPasses.h

```diff
@@ -20,14 +20,33 @@
 
 namespace glow {
 
+/// Define an enum to identify all FunctionPasses.
+enum class FunctionPassID {
+#define FUN_PASS(PASS_NAME) PASS_NAME,
+#include "FunctionPasses.def"
+};
+
+/// Declare all FunctionPass classes.
 #define FUN_PASS(PASS_NAME)                                                    \
   class PASS_NAME : public FunctionPass {                                      \
   public:                                                                      \
     bool run(Function *F) override;                                            \
     llvm::StringRef getName() const override { return #PASS_NAME; }            \
+    FunctionPassID getID() const override {                                    \
+      return FunctionPassID::PASS_NAME;                                        \
```

Comment:
is there any enforcement/check that pass names are unique?

Reply:
They're used inside an enum, so if there's a repeat name it should cause a compilation error.